lib/services/registered_addons.pm

```diff
@@ -0,0 +1,83 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Check registered system and addons status
+#
+# Maintainer: Yutao Wang <yuwang@suse.com>
+
+package services::registered_addons;
+use base 'opensusebasetest';
+use testapi;
+use utils;
+use strict;
+use warnings;
+use version_utils 'is_sle';
+use registration 'get_addon_fullname';
+
+sub zypper_lr {
+    my ($search) = @_;
+    $search //= '';
+    if ($search ne '') {
+        assert_script_run("zypper lr --uri | grep -i " . $search);
+    }
+    else {
+        assert_script_run "zypper lr --uri";
+    }
+}
+
+sub check_registered_system {
+    my ($system) = @_;
+    my $version = get_var('SLE_PRODUCT') . $system;
+    zypper_lr($version);
+}
+
+sub check_registered_addons {
+    my ($addonlist) = @_;
+    $addonlist //= get_var('SCC_ADDONS');
+    my @addons = grep { defined $_ && $_ } split(/,/, $addonlist);
+    foreach my $addon (@addons) {
+        $addon =~ s/(^\s+|\s+$)//g;
+        my $name = get_addon_fullname($addon);
+        if ($name =~ /LTSS/) {
```

Comment:
Your code is ok. Just from the code style of perl, maybe you can combine these three lines in one line as:

$name = 'LTSS' if ($name =~ /LTSS/);